Jenkinsfile.mechanical

```diff
@@ -0,0 +1,33 @@
+def utils, streams
+node {
+    checkout scm
+    utils = load("utils.groovy")
+    streams = load("streams.groovy")
+}
+
+properties([
+    pipelineTriggers(streams.get_push_trigger() + [
+        // also run every 30m: we could increase the interval more if we start
+        // triggering on the relevant fedmsgs.
+        cron("H/30 * * * *")
```

Comment:
Developer pipelines don't have any cron jobs right now. Note the only buildconfig that actually gets created by `devel-up` right now is still just `${developer_prefix}fedora-coreos-pipeline`. And that one was stripped of all triggers.

Reply:
I think we're glossing over the fact that if I want to run a developer pipeline in a brand new deployment (new project in openshift in any openshift instance) you can't start with `devel-up`. You start with `oc new-app --file=manifests/pipeline.yaml --param=DEVELOPER_PREFIX=$(whoami)-`. In this case you get **everything**, which was OK in the past because we wanted everything (i.e. we used it all for our developer pipelines usecase), but now it's not the same.

Reply:
Yeah, one base issue here is that OpenShift templates don't have conditionals (I'm not sure if Helm or something else would be better here). What we can do is split `pipeline.yaml` into 3:
- `jenkins.yaml` --> S2I + worker base + PVC (and we can split that one out as well when we're ready to make it optional)
- `pipeline.yaml` --> coreos-assembler IS + fedora-coreos-pipeline BC
- `streams.yaml` --> fedora-coreos-pipeline-{production,developer,mechanical} BCs

So then, the workflow would be `oc new-app --file=manifests/jenkins.yaml` and then straight to `./devel-up`, which instantiates `pipeline.yaml`. We could then also have e.g. a `--streams` switch if a developer wants to instantiate streams BCs as well.